### PR TITLE
feat(inspector): instrument wait for event

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: src/client/waiter.ts [hunk 1]
```diff
@@ -15,15 +15,37 @@
...
 import { EventEmitter } from 'events';
-import { rewriteErrorMessage } from '../utils/stackTrace';
+import { captureStackTrace, rewriteErrorMessage } from '../utils/stackTrace';
 import { TimeoutError } from '../utils/errors';
-import { debugLogger } from '../utils/debugLogger';
+import { createGuid } from '../utils/utils';
+import { Channel, WaitForEventInfo } from '../protocol/channels';
+import { ChannelOwner } from './channelOwner';
+
+export interface ChannelWithWaitForEventCall extends Channel {
```
Comment (dgozman) [inline src/client/waiter.ts:24]: Seems unused.

File: src/protocol/protocol.yml [hunk 2]
```diff
@@ -14,18 +14,39 @@
 
 
 # This object can be send with any rpc call in the "metadata" field.
```
Comment (dgozman) [inline src/protocol/protocol.yml:16]: This comment is now out of place.

File: src/client/waiter.ts [hunk 3]
```diff
@@ -64,7 +87,7 @@ export class Waiter {
 
   log(s: string) {
     this._logs.push(s);
-    debugLogger.log('api', s);
+    this._channelOwner._waitForEventInfoLog(this._waitId, s);
```
Comment (dgozman) [inline src/client/waiter.ts:90]: This makes me think whether we can now remove api call logging from the client and emit it from the server? We should have all the information already.

**Review by dgozman — APPROVED** [review]

File: src/client/waiter.ts (hunk 3 above)
Comment (pavelfeldman) [reply src/client/waiter.ts:90]: Yes, that's the goal. We need to add api call name to metadata for that.

**Pull request merged**